sdk/storage/azure_storage_blob/src/clients/blob_container_client.rs

```diff
@@ -59,6 +66,34 @@ impl GeneratedBlobContainerClient {
             .per_call_policies
             .push(storage_headers_policy);
 
+        #[cfg(feature = "azurite")]
+        let per_retry_policies = {
+            let use_azurite = std::env::var("AZURE_STORAGE_USE_AZURITE")
+                .map(|v| v.eq_ignore_ascii_case("true"))
+                .unwrap_or(false);
```

Comment:
This is an example implementation that work in the following manner:

- Pre-condition: `azurite` feature flag is enabled
- If enabled:

1. If `AZURE_STORAGE_USE_AZURITE` env. variable is set --> use [well-known shared key credentials](https://learn.microsoft.com/en-us/azure/storage/common/storage-connect-azurite?tabs=blob-storage#use-a-well-known-storage-account-and-key) and this enables Azurite over HTTP
2. If unset, then allow for OAuth like usual (ie. you want to use HTTPS over Azurite)